### PR TITLE
chore: update to `near_sdk` v5.1.0

### PR DESCRIPTION
**Pull request by dj8yfo**

### CODE REVIEW

File: contracts/self-updates/update/README.md [hunk 1]
```diff
@@ -5,43 +5,61 @@ in the `PostedMessage` structure.
...
-1. Deploys in on itself.
+1. Deploys it on itself.
 2. Calls the `migrate` method on itself.
 
 Lets call `update_contract` passing the new code ([./src](./src/)) using the [`manager-account`](../base/README.md#1-build-and-deploy-the-contract).
 
 ```bash
-# run from project-root/contracts
-NEW_CONTRACT_BYTES=`cat ./target/wasm32-unknown-unknown/release/self_update.wasm | base64`
-near call <dev-account> update_contract "$NEW_CONTRACT_BYTES" --base64 --accountId <manager-account> --gas 300000000000000
```
Comment (dj8yfo) [inline contracts/self-updates/update/README.md:31]: `NEAR CLI` example command was removed for this case, 
as it was a similar error both for `zsh` and `bash`:
```bash
zsh: argument list too long: near
```

probably that may be caused by arg string being too long (https://unix.stackexchange.com/questions/606847/argument-list-too-long-in-zsh-with-any-command):
```bash
❯ echo $NEW_CONTRACT_BYTES | wc -c
151877
```

**Comment by frol** [comment]
@gagdiez Given that we have only Rust contracts here, I would say it is safe to completely migrate away from JS CLI to near.cli.rs and cargo-near, and make it fully appealing to Rust devs and avoid confusion with extra instructions for JS CLI. Do you have objections?

**Comment by gagdiez** [comment]
@frol the CLI is independent of the language (since it is a CLI), so we could do as in the rest of the docs and add both commands. Otherwise, we would force people with the classic CLI to install the second one, which would collide the names and generate more confusion that good just for this example

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -1,22 +1,57 @@
...
-        uses: actions/setup-node@v2
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          target: wasm32-unknown-unknown
-      - name: Install modules
-        run: yarn
-      - name: Run tests
-        run: yarn test
```
Comment (gagdiez) [inline .github/workflows/tests.yml:22]: no need to install rust, you can simply call `cargo test`, see https://github.com/near-examples/hello-near-examples/blob/main/.github/workflows/tests-rs.yml

File: basic-updates/update/src/lib.rs [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
 use near_sdk::json_types::U128;
-use near_sdk::serde::Serialize;
-use near_sdk::{env, near_bindgen, AccountId, Balance};
+
+use near_sdk::{env, AccountId, NearToken};
 
 mod migrate;
 
-const POINT_ONE: Balance = 100_000_000_000_000_000_000_000;
+const POINT_ONE: NearToken = NearToken::from_yoctonear(100_000_000_000_000_000_000_000);
```
Comment (gagdiez) [inline basic-updates/update/src/lib.rs:10]: maybe we can transform this to `from_milinear`? same for all files

**Comment by gagdiez** [comment]
OK, I think this is good to merge, all my comments are minor and I can easy tackle them. Thank you @dj8yfo for the contribution, very much appreciated!

**Pull request merged**